gatorgrade/input/set_up_shell.py

```diff
@@ -0,0 +1,32 @@
+"""Set-up the shell commands."""
+import os, sys
+
+
+def run_setup(front_matter):
+    """Run the shell set up commands and exit the program if a command fails.
+
+    Args:
+        front_matter: A dictionary whose 'setup' key contains the set up commands
+        as a multi-line string.
+
+    """
+    # If setup does not exist in the front matter
+    if front_matter.get("setup") is not None:
+        print("Running set up commands...")
+        for line in front_matter["setup"].splitlines():
+            # Trims the white space
+            command = line.strip()
+            # Executes the command
+            exit_status = os.system(command)
```

Comment:
I don't mind the way you run the command here, but there is a way you can improve it if you're interested. Check out this article on the [different ways you can run a system command in python](https://janakiev.com/blog/python-shell-commands/).

Reply:
Thanks Noor, I saw that I can use the subprocess module because it is more versatile.

Reply:
@tuduun @noorbuchi I'm not sure that `subprocess` is actually the way to go here, since that would require that we do at least some of the shell-splitting (which you should definitely try and avoid, it's full of special edge-cases and complicated logic). `os.system` or `os.popen` (if we want to hide the output) would be the two choices I would recommend considering.

Reply:
@tuduun @noorbuchi Ah, although it's not mentioned in the link @noorbuchi provided, `subprocess`'s functions do support "shell" arguments if you set `shell=true` in the function call; you can see the specific documentation [here](https://docs.python.org/3/library/subprocess.html#frequently-used-arguments).